Content.Server/GameObjects/EntitySystems/Click/InteractionSystem.cs

```diff
@@ -561,9 +676,98 @@ public void Interaction(IEntity user, IEntity weapon, IEntity attacked, GridCoor
             foreach (var attackBy in attackBys)
             {
                 if (attackBy.AttackBy(attackByEventArgs))
-                {
                     // If an AttackBy returns a status completion we finish our attack
                     return;
+            }
+
+            // We handle specific tools AttackBy here.
+            if (weapon.TryGetComponent<ToolComponent>(out var tool))
```

Comment:
This should absolutely not be inside `InteractionSystem`.

Reply:
Done.